test/e2e/addon/addon.go

```diff
@@ -83,3 +98,75 @@ func (a Addon) WaitUtilActive(ctx context.Context, client *eks.Client, logger lo
 		}
 	}
 }
+
+func (a Addon) CreateAddon(ctx context.Context, eksClient *eks.Client, k8s clientgo.Interface, logger logr.Logger) error {
+	if err := a.Create(ctx, eksClient, logger); err != nil {
+		return err
+	}
+
+	if err := a.WaitUntilActive(ctx, eksClient, logger); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (a Addon) Delete(ctx context.Context, client *eks.Client, logger logr.Logger) error {
+	logger.Info("Delete cluster add-on", "ClusterAddon", a.Name)
+
+	params := &eks.DeleteAddonInput{
+		ClusterName: &a.Cluster,
+		AddonName:   &a.Name,
+	}
+
+	_, err := client.DeleteAddon(ctx, params)
+
+	if err == nil || errors.IsType(err, &types.ResourceNotFoundException{}) {
+		// Ignore if add-on doesn't exist
+		return nil
+	}
+	return err
+}
+
+func (a Addon) FetchLogs(ctx context.Context, k8s clientgo.Interface, logger logr.Logger) error {
```

Comment:
imo this method does fit the addon abstraction. We are representing an eks add-on. This add-on can be created, deleted and has a status that can change overtime. So the other methods make sense.

But why fetch logs? I understand most add-ons deploy pods, but that's not part of the add-on concept. This is a bit academical, but I fear if we break the abstraction like this, we just open the doors for this to become a dumping ground.

It seems like the only things you need from the add-on resource is name and namespace. Why not just make that arguments to the method and free it from the add-on abstraction?

Reply:
Moved it to `test/e2e/kubernetes/pod.go`